src/irr/asset/CPLYMeshFileLoader.cpp

```diff
@@ -597,17 +592,41 @@ bool CPLYMeshFileLoader::genVertBuffersForMBuffer(asset::ICPUMeshBuffer* _mbuf,
 
 	const uint32_t vertexSize = std::accumulate(sizes, sizes + 4, static_cast<size_t>(0));
 
+	// TODO default shader (uber shader isn't a goal since there will be only one vertex and fragment shader)
+	auto pushConstantRange = core::make_refctd_dynamic_array<core::smart_refctd_dynamic_array<SPushConstantRange>>(2);
+	pushConstantRange->operator[](0).stageFlags = ISpecializedShader::E_SHADER_STAGE::ESS_VERTEX;
+	pushConstantRange->operator[](0).offset = 0; //TODO
+	pushConstantRange->operator[](0).size = 0; //TODO
+
+	pushConstantRange->operator[](1).stageFlags = ISpecializedShader::E_SHADER_STAGE::ESS_FRAGMENT;
+	pushConstantRange->operator[](1).offset = 0; //TODO
+	pushConstantRange->operator[](1).size = 0; //TODO
+
+	auto mbPipelineLayout = core::make_smart_refctd_ptr<ICPUPipelineLayout>(pushConstantRange->begin(), pushConstantRange->end()); 
```

Comment:
you can add a default pipeline layout and shader to the asset manager's built-ins for a single texture material with vertex color and normals

Reply:
okay

Reply:
so should I create `CGLSL_PLYBuiltinIncludeLoader` or something like that?

Reply:
just insert it into cache?

Reply:
@devshgraphicsprogramming what do you want me to do in fragment shader with normals?

Reply:
create these defaults in AssetManager.cpp (next to the others we have there already) and give them nice and sane paths when inserting into cache.